### PR TITLE
refact(skymp5-front): rewrite SkyrimInput in TS

### PR DESCRIPTION
**Pull request by JustCaptcha**
Closes: [476](https://github.com/skyrim-multiplayer/skymp/issues/476).

### CODE REVIEW

File: skymp5-front/src/interfaces/buttons.ts [hunk 1]
```diff
@@ -5,6 +5,14 @@ export interface SkyrimHintProps {
     left: boolean;
 }
 
+export interface SkyrimInputProps extends DefaultButtonComponentProps {
+    labelText: string;
+    type: React.HTMLInputTypeAttribute;
+    initialValue: string | number | readonly string[];
+    placeholder: string;
+    onInput?: (any?: any) => void;
```
Comment (Pospelove) [inline skymp5-front/src/interfaces/buttons.ts:13]: there can be concrete type
`onInput?: (text: string) => void | () => void | (isChecked: boolean) => void`

**Review by Pospelove — CHANGES_REQUESTED** [review]

File: skymp5-front/src/interfaces/buttons.ts (hunk 1 above)
Comment (Pospelove) [reply skymp5-front/src/interfaces/buttons.ts:13]: So there is still `any`

**Review by Pospelove — CHANGES_REQUESTED** [review]

**Comment by nic11** [comment]
@JustCaptcha @Pospelove `refact`, I guess?

**Comment by vladdenisov** [comment]
Will we use `onInput` on buttons? Or we will stay with `onClick`?

**Review by Pospelove — CHANGES_REQUESTED** [review]
The PR isn't buildable. Please check build errors in CI

**Comment by Pospelove** [comment]
https://github.com/skyrim-multiplayer/skymp/runs/4272767380?check_suite_focus=true#step:13:462

**Comment by vladdenisov** [comment]
> The PR isn't buildable. Please check build errors in CI

one of the reasons is that we used `onClick` property for buttons and it was changed here to `onInput`

**Review by Pospelove — APPROVED** [review]

**Pull request merged**